src/python/pants/build_graph/build_graph.py

```diff
@@ -344,14 +366,36 @@ def walk_transitive_dependency_graph(self, addresses, work, predicate=None, post
       out of the closure.  If it is given, any Target which fails the predicate will not be
       walked, nor will its dependencies.  Thus predicate effectively trims out any subgraph
       that would only be reachable through Targets that fail the predicate.
-    :param function leveled_predicate: Behaves identically to predicate, but takes the depth of the
+    :param function dep_predicate: Takes two parameters, the current target and the dependency of
+      the current target. If this parameter is not given, no dependencies will be filtered
+      when traversing the closure. If it is given, when the predicate fails, the edge to the dependency
+      will not be expanded.
+    :param function leveled_predicate: Deprecated. Behaves identically to predicate, but takes the depth of the
       target in the search tree as a second parameter, and it is checked just before a dependency is
       expanded.
     """
+    deprecated_conditional(
+      lambda: leveled_predicate is not None,
+      '1.6.0.dev0',
```

Comment:
Will need to bump this to 1.7.0.dev0 I think.

Reply:
On it.